posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc

```diff
@@ -0,0 +1,145 @@
+    <type>zip</type>
+</dependency>
+----
+Or for link:{url-prefix}/guides/gradle-intro.html[Gradle]:
+[source,gradle]
+----
+dependencies {
+    libertyRuntime group: 'io.openliberty', name: 'openliberty-runtime', version: '[20.0.0.9,)'
+}
+----
+Or if you're using Docker:
+[source]
+----
+FROM open-liberty
+----
+//end::run[]
+Or take a look at our link:{url-prefix}/downloads/[Downloads page].
+[link=https://stackoverflow.com/tags/open-liberty]
+image::img/blog/blog_btn_stack.svg[Ask a question on Stack Overflow, align="center"]
+
+//tag::features[]
+[#GraphQLAPIs]
+== Experiment with a third-party GraphQL client API
+
+MicroProfile GraphQL has only been available in Open Liberty for a few months now, and it is already a hit! That said, there are lots of things we'd like to do to improve it - and make it more complete. One of those things is a Client API. While the official Client API is not expected until the next release of MicroProfile GraphQL, you can now experiment with the type-safe SmallRye GraphQL Client API. SmallRye is the underlying implementation of MicroProfile GraphQL, and you can access some of its above-and-beyond-the-spec features by adding the "third-party" API type visibility to your application, like so:
+[source,xml]
+----
+<application name="MyGraphQLApp" location="MyGraphQLApp.war">
+    <classloader apiTypeVisibility="+third-party"/>
+</application>
+----
+Now you can access SmallRye GraphQL APIs like the type-safe client!  Note that these APIs may change in future releases as SmallRye is constantly evolving.  For more information on the SmallRye GraphQL project, please visit https://github.com/smallrye/smallrye-graphql
+The SmallRye GraphQL Client APIs are very similar to https://github.com/eclipse/microprofile-rest-client[MicroProfile Rest Client] in that it uses an interface to invoke remote methods in a type-safe manner. For example, suppose we want a client that can invoke a query of all super heroes in a given location.  We would create a query interface like this:
+[source,java]
+----
+@GraphQlClientApi
+interface SuperHeroesApi {
+    List<SuperHero> allHeroesIn(String location); // (1)
+}
+----
+Where `SuperHero` on the client side looks like this:
+[source,java]
+----
+class SuperHero {
+    private String name;
+    private List<String> superPowers;
+}
+----
+The `SuperHero` entity might contain dozens more fields, but if we're only interested in the hero's name and super powers, then we only need those two fields in our client-side class.  Now we can invoke the query with some code like this:
+[source,java]
+----
+SuperHeroesApi api = GraphQlClientBuilder.newBuilder().build(SuperHeroesApi.class);
+List<SuperHero> heroesOfNewYork = api.allHeroesIn("NYC");
+----
+Remember that this client API is not official, but the official API (in MicroProfile GraphQL 1.1) will be based on this - think of this as a sneak peak!
+
+[#GraphiQL]
+== Built-in GraphiQL user interface for faster queries and mutations
```

Comment:
```suggestion
== Built-in GraphiQL user interface for faster development of queries and mutations
```

The UI is intended to make it easier and faster for developers to create and test queries and mutation.

Reply:
Done